csrc/cpu/neighbor_sample_cpu.cpp

```diff
@@ -409,22 +428,15 @@ hetero_temporal_neighbor_sample_cpu(
     const c10::Dict<node_t, torch::Tensor> &input_node_dict,
     const c10::Dict<rel_t, vector<int64_t>> &num_neighbors_dict,
     const c10::Dict<node_t, torch::Tensor> &node_time_dict,
-    const int64_t num_hops, const bool replace, const bool directed) {
+    const int64_t num_hops, const bool replace) {
 
-  if (replace && directed) {
+  if (replace) {
+    // directed = True for temporal sampling
     return hetero_sample<true, true, true>(
         node_types, edge_types, colptr_dict, row_dict, input_node_dict,
         num_neighbors_dict, node_time_dict, num_hops);
-  } else if (replace && !directed) {
-    return hetero_sample<true, false, true>(
-        node_types, edge_types, colptr_dict, row_dict, input_node_dict,
-        num_neighbors_dict, node_time_dict, num_hops);
-  } else if (!replace && directed) {
-    return hetero_sample<false, true, true>(
-        node_types, edge_types, colptr_dict, row_dict, input_node_dict,
-        num_neighbors_dict, node_time_dict, num_hops);
   } else {
-    return hetero_sample<false, false, true>(
+    return hetero_sample<false, true, true>(
```

Comment:
Let' s throw an error in case `temporal` but `!directed`?

Reply:
I removed the directed option for now, and by default it's always directed. In future we could think of extending the directed option for temporal sampling.

Reply:
I would be more comfortable with raising an error here. Ignoring `directed` option is dangerous here since this is typically used whenever the GNN has more layers than we sample hops.